jung-visualization/src/main/java/edu/uci/ics/jung/visualization/renderers/BasicVertexLabelRenderer.java

```diff
@@ -112,7 +109,16 @@ public void labelVertex(
     } else {
       p = getAnchorPoint(bounds, d, position);
     }
-    g.draw(component, renderContext.getRendererPane(), p.x, p.y, d.width, d.height, true);
+
+    Paint oldPaint = component.getForeground();
+    Paint fillPaint = renderContext.getVertexLabelDrawPaintTransformer().apply(v);
+    if (fillPaint != null) {
+      component.setForeground((Color) fillPaint);
+      g.draw(component, renderContext.getRendererPane(), p.x, p.y, d.width, d.height, true);
+      component.setForeground((Color) oldPaint);
+    } else {
+      g.draw(component, renderContext.getRendererPane(), p.x, p.y, d.width, d.height, true);
+    }
```

Comment:
see above comment about label color. Here is where the label color is specifically accessed instead of just using whatever the container (the vv) has as its foreground color.